src/main/java/com/palantir/refreshable/DefaultRefreshable.java

```diff
@@ -45,20 +47,32 @@
  */
 final class DefaultRefreshable<T> implements SettableRefreshable<T> {
     private static final Logger log = LoggerFactory.getLogger(DefaultRefreshable.class);
-    /**
-     * Every ten times refreshable.map is called without an update we must purge SelfRemovingMapSubscriber instances
-     * whose children have been reaped.
-     */
-    private static final int CLEAN_THRESHOLD = 10;
+    private static final Cleaner REFRESHABLE_CLEANER = Cleaner.create();
 
     private static final int WARN_THRESHOLD = 1000;
 
-    @GuardedBy("this")
-    private final Set<Consumer<? super T>> orderedSubscribers = new LinkedHashSet<>();
+    /** Subscribers are updated in deterministic order based on registration order. This prevents a class
+     * of bugs where a listener on a refreshable uses a refreshable mapped from itself, and guarantees the child
+     * mappings will be up-to-date before the listener is executed, as long as the input mapping occurred before
+     * the subscription. While we strongly recommend against this kind of dependency, it's complicated to detect
+     * in large projects with layers of indirection.
+     * <p>
+     * Consider the following:
+     * <pre>{@code
```

Comment:
That is perfect: nit: do you need ``<pre> {@code``? wouldn't either one of these suffice?

Reply:
Not quite, the combination of the two preserves formatting: https://reflectoring.io/howto-format-code-snippets-in-javadoc/